src/v2/Apps/Partner/Components/Overview/NearbyGalleryCard.tsx

```diff
@@ -0,0 +1,142 @@
+  city: string | null
+}
+
+function normalizeCityName(city: string) {
+  return capitalize(city.trim())
+}
+
+function getLocation(cities: Array<string>, preferredCity: string | null) {
+  let location
+
+  if (cities.length > 0) {
+    const normalizedPreferredCity =
+      preferredCity && normalizeCityName(preferredCity)
+
+    if (cities.some(c => c === normalizedPreferredCity)) {
+      location = normalizedPreferredCity
+    } else {
+      location = cities[0]
+    }
+
+    if (cities.length > 1) {
+      location += ` & ${cities.length - 1} other location`
+
+      if (cities.length > 2) {
+        location += "s"
+      }
+    }
+  }
+
+  return location
+}
+
+const NearbyGalleryCard: React.FC<NearbyGalleryCardProps> = ({
+  partner,
+  city,
+  ...rest
+}) => {
+  const { user } = useSystemContext()
+
+  if (!partner) {
+    return null
+  }
+
+  const { name, slug, profile, type, locationsConnection } = partner
+  const canFollow = type !== "Auction House"
+  const image = partner?.profile?.image?.cropped
+  const partnerHref = `/partner2/${slug}`
+  const cities = uniq(
+    compact(
+      locationsConnection?.edges?.map(location =>
+        location?.node?.city
+          ? normalizeCityName(location?.node?.city)
+          : location?.node?.displayCountry
+      )
+    )
+  )
+  const location = getLocation(cities, city)
+
+  return (
+    <Box {...rest}>
+      <ResponsiveBox
+        bg="black10"
+        aspectHeight={3}
+        aspectWidth={4}
+        maxWidth="100%"
+      >
+        <RouterLink to={partnerHref}>
+          {image && (
+            <Image
+              lazyLoad
+              src={image.src}
+              srcSet={image.srcSet}
+              width="100%"
+              height="100%"
+            />
+          )}
+        </RouterLink>
+      </ResponsiveBox>
+      <Flex justifyContent="space-between" mt={1}>
+        <Box>
+          <RouterLink noUnderline to={partnerHref}>
+            <Text variant="subtitle">{name}</Text>
+            {location && <Text color="black60">{location}</Text>}
+          </RouterLink>
+        </Box>
+        {canFollow && !!profile && (
+          <FollowProfileButtonFragmentContainer
+            profile={profile}
+            user={user}
+            contextModule={ContextModule.partnerHeader}
+            buttonProps={{
+              size: "small",
+              variant: "secondaryOutline",
+            }}
+          />
+        )}
+      </Flex>
+    </Box>
+  )
+}
+
+export const NearbyGalleryCardFragmentContainer = createFragmentContainer(
+  NearbyGalleryCard,
+  {
+    partner: graphql`
+      fragment NearbyGalleryCard_partner on Partner {
+        name
+        slug
+        type
+        profile {
+          image {
+            cropped(height: 300, width: 400, version: "wide") {
+              src
+              srcSet
+            }
+          }
+          ...FollowProfileButton_profile
+        }
+        locationsConnection(first: 20) {
```

Comment:
20 seems like a lot to pull for this little card-- can it be like `5` instead?

Reply:
locationsConnection using here to show this line
![image](https://user-images.githubusercontent.com/79979820/118991262-3fb02300-b98c-11eb-9d98-b27e68a25dc0.png)

The city at the card should be equal to the city from the first partner location, so if I load a small number of locations there may not be a preferred city and I'll show the wrong city.

What do you think if we change the line from ‘CityName & # other locations’ to '# locations'? I think in this case we can use just `totalCount` from the `locationsConnection`.